### PR TITLE
Added en passant capture logic

### PR DESCRIPTION
**Pull request by jamesfoster85**
- Added en_passant column to pieces table.
- Added logic to the `move!` method:
  - Moving a pawn forward 2 squares changes its en_passant value to `true`.
  - The value is then changed back to `false` after the opponent's next move, whether or not that next move is to capture the pawn en passant.
  - Only pawns can *capture* en passant, and only pawns can *be captured* en passant; an en_passant value of `true` indicates that this is a pawn that can *be captured* en passant.
- Added logic on the Pawn model to determine whether the pawn can *capture* en passant.
  - Takes into account both the presence of a `true` flag on an opponent's pawn and the relative positions of the pieces.
  - Makes en passant a `legal_move`.
- Added additional logic to the `move!` method so that after an en passant move is made, the `capture_opponent` method is correctly applied to the opponent's pawn.

### CODE REVIEW

**Comment by lasley** [comment]
Conflicts yay!

**Comment by jamesfoster85** [comment]
Resolved

File: app/models/pawn.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@ def legal_moves
     unobstructed_squares.select do |s|
       (forward_one?(s) && !occupied?(s)) ||
         (forward_two?(s) && !occupied?(s) && not_moved) ||
-        (captures_diagonally?(s) && occupied?(s))
+        (captures_diagonally?(s) && occupied?(s)) ||
+        (captures_diagonally?(s) && en_passant_capture?(s))
```
Comment (lasley) [inline app/models/pawn.rb]: Might want to save the returns of these methods to an intermediary variable in order to be evaluated. 

Calling the method every time means that a bunch of logic is being unnecessary called as well, and would only be worth it if the game were changing out from underneath the conditionals.

**Comment by lasley** [comment]
:+1:

**Pull request merged**